docs/concurrency.md

```diff
@@ -89,24 +91,26 @@ To formalize the difference between general and acyclic dataflow, we note that a
-The types `Model`, `View`, and `Input` stand for data that represent the type of the data model of the application; the type of the entire view (all windows) shown on the screen, and the possible input event that the user might create (including keyboard and mouse).
-The three streams are defined recursively: the `View` is a function of the `Model`; the `Stream[Model]` is a function of `Stream[Input]` (user input events update the model); and `Input` also depends on `View` since the `View` determines what control elements are shown on the screen at the time, and thus what input events the user can create.
+The types `Model`, `View`, and `Input` stand for data that represent the type of the data model of the application; the type of the entire view (all windows) shown on the screen; and all the possible input events that the user might create (including keyboard and mouse).
+The three streams are defined recursively: the `View` is a function of the `Model`; the `Stream[Model]` is a function of `Stream[Input]` (some input events will update the model); and `Input` also depends on `View` since the `View` determines which control elements are shown on the screen at any time, and thus determines what input events the user can create at that time.
 
 The streaming frameworks that do not support a recursive definition of streams fall into the acyclic dataflow class.
 
 ## Level 4: General concurrency
 
 Finally, we consider the general concurrency problems.
 The main task in this class of problems is to manage several computations that run concurrently in unknown order and interact in arbitrary ways.
-For instance, one computation may at some point stop and wait until another computation computes a certain result, and then examine that result to decide whether to continue its own thread of computation or wait further.
+For instance, one computation may at some point stop and wait until another computation computes a certain result, and then examine that result to decide whether to continue its own thread of computation, or wait further, or create new computation threads.
 
 The main difficulty here is to ensure that different processes are synchronized in the desired manner.
 
 Frameworks such as Akka Actors, Go coroutines/channels, and the Java concurrency primitives (`Thread`, `wait/notify`, `synchronized`) are all Level 4 concurrency frameworks.
 The chemical machine (known in the academic world as "join calculus") is also a Level 4 framework.
 
-The typical task that requires this level of concurrency is implementing an operating system.
-The "dining philosophers" problem is also an example of a concurrency task that cannot be implemented by any concurrency framework other than a Level 4 framework.
+The typical task that requires this level of concurrency is implementing an operating system where many running processes can synchronize and communicate with each other in arbitrary ways.
```

Comment:
What about "a typical use case" instead of "the typical task"? I was thinking about the use cases, not about specific frameworks here. Frameworks are discussed later, for each level.